src/inet/TCPEndPoint.cpp

```diff
@@ -2716,7 +2757,7 @@ CHIP_ERROR TCPEndPoint::DoClose(CHIP_ERROR err, bool suppressCallback)
         //
         if (oldState != State::kReady && oldState != State::kBound)
         {
-            DeferredFree(kReleaseDeferralErrorTactic_Ignore);
+            Release();
```

Comment:
The comment starting on line 2709 says we can't release directly in this context.  Did something change such that we can now safely do this?

Reply:
Ah, I think I understand.  So our new strategy is that any code enqueuing events will have called Retain.  Thus is safe to call Release directly in all cases, which will not decrement the reference count to zero when that's the behavior we need.

Got it.  Please update the above comment at line 2709 to reflect.